src/components/CohortBuilder/Summary/SurvivalChart.js

```diff
@@ -181,18 +183,68 @@ const StyledSurvivalPlot = styled(SurvivalPlot)`
 class SurvivalChart extends React.Component {
   constructor(props) {
     super(props);
+
     this.state = {
       tooltip: {
         donor: {},
         x: 0,
         y: 0,
         isVisible: false,
       },
+      isLoading: true,
+      data: {},
     };
     this.handleMouseEnterDonors = this.handleMouseEnterDonors.bind(this);
     this.handleMouseLeaveDonors = this.handleMouseLeaveDonors.bind(this);
   }
 
+  // Fix problem where fetches are loaded into state in the wrong order due to slow running large queries
+  queryCacheMap = {};
+  cachedFetch = () => {
+    const { api, sqon } = this.props;
+    const hash = md5(JSON.stringify(sqon));
```

Comment:
@hlminh2000 We already have a hashing library in place - solution for your QueriesResolver as well!